### PR TITLE
Topic/ap 271 update currencucloud library

### PR DESCRIPTION
**Pull request by oleg-sviridov**
AP-271 updated to .net8, changed DateTime to DateTimeOffset in entities to resolve timezone issues, fixed path referencing in tests, updated tests to work with DateTimeOffset

### CODE REVIEW

**Review by baklanser — APPROVED** [review]

File: Source/CurrencyCloud.Tests/Mock/Data/ReportRequests.cs [hunk 1]
```diff
@@ -80,12 +80,12 @@ static class ReportRequests
...
+            CreatedAt = DateTimeOffset.Parse("2018-01-01T12:34:56+00:00"),
             Id = "0ec2bc31-5dd8-4029-9491-6c4c0598a1cc",
             Status = "processing",
             FailureReason = null,
             ExpirationDate = null,
-            UpdatedAt = new DateTime(2018, 01, 01, 12, 34, 56)
+            UpdatedAt = DateTimeOffset.Parse("2018-01-01T12:34:56+00:00")
         };
     }
 }
```
Comment (vsevolod-mozgovoi-am) [inline Source/CurrencyCloud.Tests/Mock/Data/ReportRequests.cs:91]: EOF

File: Source/CurrencyCloud.Tests/Mock/Data/Transactions.cs [hunk 2]
```diff
@@ -10,11 +10,11 @@ static class Transactions
...
+            ValueDate = DateTimeOffset.Parse("2018-01-01T00:00:00+00:00"),
             Sender = "FR7615589290001234567890113, CMBRFR2BARK, Debtor, FR, Centre ville",
             ReceivingAccountIban = "GB99OXPH94665099600083",
-            CreatedAt = new DateTime(2018, 01, 01, 12, 34, 56),
-            UpdatedAt = new DateTime(2018, 01, 01, 12, 34, 56)
+            CreatedAt = DateTimeOffset.Parse("2018-01-01T12:34:56+00:00"),
+            UpdatedAt = DateTimeOffset.Parse("2018-01-01T12:34:56+00:00")
         };
     }
 }
```
Comment (vsevolod-mozgovoi-am) [inline Source/CurrencyCloud.Tests/Mock/Data/Transactions.cs:20]: EOF

File: Source/CurrencyCloud/CurrencyCloud.csproj [hunk 3]
```diff
@@ -1,173 +1,20 @@
...
-  </Target>
-  <!-- To modify your build process, add your task inside one of the targets below and uncomment it. 
-       Other similar extension points exist, see Microsoft.Common.targets.
-  <Target Name="BeforeBuild">
-  </Target>
-  <Target Name="AfterBuild">
-  </Target>
-  -->
+
 </Project>
```
Comment (vsevolod-mozgovoi-am) [inline Source/CurrencyCloud/CurrencyCloud.csproj:20]: EOF

File: Source/CurrencyCloud/Environment/Platform.cs [hunk 4]
```diff
@@ -1,82 +1,19 @@
...
         public static readonly string Version;
+
+        static Platform()
+        {
+            Version = Assembly.GetEntryAssembly()?.GetCustomAttribute<TargetFrameworkAttribute>()?.FrameworkName ??
+                      ".NET";
+        }
     }
-}
+}
```
Comment (vsevolod-mozgovoi-am) [inline Source/CurrencyCloud/Environment/Platform.cs:19]: EOF

File: Source/CurrencyCloud/Environment/Platform.cs [hunk 5]
```diff
@@ -1,82 +1,19 @@
...
-                    }
-
-                    Version = ".NET " + versionNumber;
-                }
-            }
-        }
-
         /// <summary>
-        /// Gets version number of the installed Mono or .NET framework.
+        ///     Gets version number of the installed Mono or .NET framework.
```
Comment (vsevolod-mozgovoi-am) [inline Source/CurrencyCloud/Environment/Platform.cs:9]:
```suggestion
        /// Gets .NET version number.
```

**Review by vsevolod-mozgovoi-am — APPROVED** [review]

**Review by maksim-resnenko — APPROVED** [review]

**Pull request merged**